rke/templates/nginx-ingress.go

```diff
@@ -1015,9 +1019,23 @@ spec:
 {{end}}
           ports:
           - name: http
+            {{- if eq .NetworkMode "hostNetwork"}}
             containerPort: 80
+            {{- else if or (eq .NetworkMode "hostPort") (eq .NetworkMode "none")}}
+            containerPort: {{with (index .ExtraArgs "http-port")}}{{.}}{{else}}80{{end}}
+			{{- if eq .NetworkMode "hostPort"}}
+            hostPort: {{.HTTPPort}}
+			{{- end }}
+            {{- end }}
           - name: https
+            {{- if eq .NetworkMode "hostNetwork"}}
             containerPort: 443
+            {{- else if or (eq .NetworkMode "hostPort") (eq .NetworkMode "none")}}
+            containerPort: {{with (index .ExtraArgs "https-port")}}{{.}}{{else}}443{{end}}
+            {{- if eq .NetworkMode "hostPort"}}
```

Comment:
the indentations on the if matter here? just weird one above has and this one doesnt